library/farmosUtil/farmosUtil.soilDisturbanceTerminationLog.unit.cy.js

```diff
@@ -0,0 +1,630 @@
+        expect(soilDisturbanceLog.relationships.category[0].id).to.equal(
+          categoryMap.get('termination').id
+        );
+      }
+    );
+
+    // Ensure the plant asset in the termination log is archived
+    cy.get('@readSoilDisturbanceLog').then((soilDisturbanceLog) => {
+      cy.wrap(
+        farmosUtil.getPlantAsset(soilDisturbanceLog.relationships.asset[0].id)
+      ).then((updatedAsset) => {
+        expect(updatedAsset.attributes.status).to.equal('archived');
+        expect(updatedAsset.relationships.location).to.have.length(1);
+      });
+    });
+
+    // Delete all logs and plant asset
+    cleanupLogsAndAsset({
+      deleteSoilDisturbanceLog: true,
+      deletePlantAsset: true,
+    });
+  });
+
+  it(
+    'Error creating a soil disturbance termination log for specific beds',
+    { retries: 4 },
+    () => {
+      createPlantAsset(
+        'Test Plant Asset for Error Case',
+        'HERB-CILANTRO',
+        '2023-11-20'
+      );
+
+      // Assign beds to the plant asset via a movement log
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        createMovementLog(
+          plantAsset,
+          ['CHUAU', 'CHUAU-1', 'CHUAU-3'],
+          ['seeding'],
+          '2023-11-20'
+        );
+      });
+
+      // Simulate an error while creating the soil disturbance termination log
+      cy.intercept('POST', '**/api/log/activity', {
+        statusCode: 401,
+      });
+
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+          (updatedPlantAsset) => {
+            cy.wrap(
+              farmosUtil
+                .createSoilDisturbanceTerminationLog(
+                  '2023-11-20',
+                  'CHUAU',
+                  ['CHUAU-3'], // Attempt to terminate CHUAU-3
+                  updatedPlantAsset
+                )
+                .then(() => {
+                  throw new Error(
+                    'Creating soil disturbance log should have failed.'
+                  );
+                })
+                .catch((error) => {
+                  expect(error.message).to.equal(
+                    'Request failed with status code 401'
+                  );
+                })
+            );
+          }
+        );
+      });
+
+      // Recheck the plant asset to ensure its state remains intact
+      cy.get('@newPlantAsset').then((plantAsset) => {
+        cy.wrap(farmosUtil.getPlantAsset(plantAsset.id)).then(
+          (updatedAsset) => {
+            expect(updatedAsset.relationships.location).to.have.length(3); // Beds should remain intact
+            expect(updatedAsset.attributes.status).to.not.equal('archived'); // Asset should not be archived
+          }
+        );
+      });
+
+      // Delete the movement log and plant asset
+      cleanupLogsAndAsset({
+        deleteMovementLog: true,
+        deletePlantAsset: true,
+      });
+    }
+  );
+
+  it(
```

Comment:
I'm not sure we need this test or the next one.  Looks like they all test for an error on the same endpoint that would occur at the same place in the `createSoilDisturbanceTerminationLog` function.  So I think if one passes they will all pass and if one fails, they would all fail.  Though, I may be missing something here, if so please respond and elaborate on what each test is checking that is different.

Reply:
Thanks for pointing that out! I agree with you—since they all hit the same endpoint and logic in `createSoilDisturbanceTerminationLog`, if one fails, they’ll probably all fail. Honestly, I was just being overly cautious. I was thinking along the lines of testing different scenarios to make sure it works for:

- A plant asset with specific beds selected for termination.  
- A plant asset with all its beds selected for termination.  
- A plant asset with no beds at all.

But yeah, now that you mention it, it might not add much value since they all depend on the same core behavior. If you think it’s better to simplify things, I can remove or adjust these. Let me know what you think!